extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVault.java

```diff
@@ -39,6 +40,10 @@ public HashicorpVault(@NotNull HashicorpVaultClient hashicorpVaultClient, @NotNu
     public @Nullable String resolveSecret(String key) {
         var result = hashicorpVaultClient.getSecretValue(key);
 
+        if (result.failed()) {
+            monitor.warning("Failed to resolve secret: " + result.getFailureMessages());
```

Comment:
Can you include the key name, e.g. warning(format(Failed to resolve secret '%s':%s", key, result.getFailureMessage()));

Reply:
@jimmarino You are right, I added the key to the message.